Source/WarpX.cpp

```diff
@@ -977,11 +976,13 @@ WarpX::ReadParameters ()
         }
 
         constexpr int zdir = AMREX_SPACEDIM - 1;
-        if (!Geom(0).isPeriodic(zdir))
-        {
-            use_damp_fields_in_z_guard = true;
+        if (WarpX::field_boundary_lo[zdir] == FieldBoundaryType::Damped ||
+            WarpX::field_boundary_hi[zdir] == FieldBoundaryType::Damped ) {
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(
```

Comment:
You can add here
AMREX_ALWAYS_ASSERT_WITH_MESSAGE(WarpX::maxwell_solver_id == MaxwellSolverAlgo::PSATD, "Error: Damped field boundary type is only supported with PSATD");

Reply:
But this part of the code is only inside 
if (maxwell_solver_id == MaxwellSolverAlgo::PSATD)
This the Assert would be checked only for cases with PSATD. I added an else below, or I could move the check at the end of the if block as well.

Reply:
This is good (I didn't notice that this was within the if block). Though, since that is such a long if block, I would suggest adding a new if block with `if (maxwell_solver_id != MaxwellSolverAlgo::PSATD) {` instead of the `else`. This will make the code more readable.

Reply:
Thank you @dpgrote I have made this change now.